### ISSUE TITLE
std::string::size_type causing problems on Mac gcc-8.1.0

### ISSUE THREAD

**Issue by pjh40** [flang-compiler/f18#125]
Under gcc-8.1.0 (or libstdc++-8.1.0 to be more specific), the std::string::size_type is 'unsigned long'.  This causes compilation errors with the instantiation of 
```c++
template<int KIND>
typename CharacterExpr<KIND>::LengthExpr CharacterExpr<KIND>::LEN() const;
```
at expression.cc:211, because there is no Integer(unsigned long) ctor, and g++ is reluctant to convert UL to ULL.  This can be addressed with a `std::uint64_t(x.size())` in the base case of the visitor constructor, or, perhaps more robustly, adding `Integer(unsigned long)`.

**Comment by klausler** [comment]
I will add a static_cast to avoid potential conflicts between the Integer constructors of fixed size integer types (e.g., std::uint64_t) and the implementation-defined types (e.g., "unsigned long"), which I've been trying to avoid using generally.

**Comment by klausler** [comment]
Should be fixed as of commit 57e051eafa1088e83c7d1cef565add6903aed20a, please confirm when convenient.

**Comment by pjh40** [comment]
works for me, thanks for the fix!
